src/gui/windows/main.zig

```diff
@@ -18,8 +22,135 @@ const padding: f32 = 8;
 fn exitGame(_:usize) void {
 	main.Window.c.glfwSetWindowShouldClose(main.Window.window,main.Window.c.GLFW_TRUE);
 }
+
+fn deleteDirectoryRecursive(fs: std.fs.Dir, allocator: std.mem.Allocator, path: []const u8) anyerror!void {
+	var dir = try fs.openDir(path, .{.iterate = true});
+	defer dir.close();
+
+	var it = dir.iterate();
+
+	while (try it.next()) |entry| {
+		if (entry.kind == .file) {
+			try dir.deleteFile(entry.name);
+		} else if (entry.kind == .directory) {
+			try deleteDirectoryRecursive(dir, allocator, entry.name);
+		}
+	}
+
+	try fs.deleteDir(path);
+}
+
+pub fn play(_:usize) void {
+	deleteDirectoryRecursive(std.fs.cwd(), std.heap.page_allocator, ".cubyz/Cubyz-master") catch |err| {
+		std.debug.print("{}\n", .{err});
+		return;
+	};
+
+	http.fetchAndSave(std.heap.page_allocator, "https://www.github.com/PixelGuys/Cubyz/archive/master.zip", ".cubyz/master.zip") catch |err| {
+		std.debug.print("{}\n", .{err});
+		return;
+	};
+	
+	archive.decompress_zip(".cubyz/master.zip", ".cubyz") catch |err| {
+		std.debug.print("{}\n", .{err});
+		return;
+	};
+	
+	std.fs.cwd().deleteFile(".cubyz/master.zip") catch {};
+
+	const dir = std.fs.cwd().openDir(".cubyz/Cubyz-master", .{}) catch |err| {
+		std.debug.print("{}\n", .{err});
+		return;
+	};
+
+	const file = dir.openFile(".zig-version", .{}) catch |err| {
+		std.debug.print("{}\n", .{err});
+		return;
+	};
+	defer file.close();
+
+	const ver = file.readToEndAlloc(std.heap.page_allocator, 32) catch |err| {
```

Comment:
Instead of opening the file manually, you can just do
```
const dir = main.files.openDir(".cubyz/Cubyz-master") catch |err| {..};
defer dir.close();
const ver = dir.read(main.stackAllocator, ".zig-version") catch |err| {...};
defer main.stackAllocator.free(ver);
```
This essentially combines the `openFile` and `readToEndAlloc` steps.